lib/service_manager.dart

```diff
@@ -96,7 +130,12 @@ class ServiceConnectionManager {
 
       service.onServiceEvent.listen((e) {
         if (e.kind == EventKind.kServiceRegistered) {
-          methodsForService.putIfAbsent(e.service, () => []).add(e.method);
+          registeredMethodsForService
+              .putIfAbsent(e.service, () => [])
+              .add(e.method);
+          final StreamController<bool> streamController =
+              _getServiceRegistrationController(e.service);
+          streamController.add(true);
```

Comment:
should we be adding true to the stream if this isn't the first time the serviceRegistered event was called for this stream?
I would expect we want to only add this to the stream in the putIfAbsent case.

Reply:
It shouldn't negatively affect anything if we do because we aren't sending a different boolean value. But only doing it in the putIfAbsent case could save the stream listener some work.